code/modules/modular_computers/hardware/gprs_card.dm

```diff
@@ -0,0 +1,126 @@
+// Since we can't assure that the PDA program is running at all times,
+// We need to do this in "hardware"
+//This is the definition of overdesigned, but the way it used to work was also awful, so eat shit.
+
+#define GPRS_ENABLED TRUE
+#define GPRS_DISABLED FALSE
+
+/obj/item/computer_hardware/network_card/packetnet
+	name = "\improper GPRS network card"
+	desc = "A cheap network card with an attached GPRS Modem. Supports NTNet and GPRS Messaging."
+	var/gprs_frequency = FREQ_COMMON
+	var/datum/radio_frequency/radio_connection
+
+	/// Stores up to [queue_max] messages until the PDA asks for them.
+	VAR_PRIVATE/list/datum/signal/packet_queue
+	/// Maximum amount of messages held by [packet_queue]
+	var/queue_max = 5
+	var/radio_state = GPRS_ENABLED
+
+/obj/item/computer_hardware/network_card/packetnet/Initialize(mapload)
```

Comment:
Destroy() needs to call set_radio_state(null) by the looks.

Reply:
Destroy directly removes the obj from the relevant lists.
```
/obj/item/computer_hardware/network_card/packetnet/Destroy()
	. = ..()
	SSpackets.remove_object(src, gprs_frequency, RADIO_PDAMESSAGE)
	radio_connection = null
	packet_queue.Cut()
```